src/librustc_ast/util/comments.rs

```diff
@@ -61,88 +76,45 @@ pub fn doc_comment_style(comment: &str) -> ast::AttrStyle {
     }
 }
 
-pub fn strip_doc_comment_decoration(comment: &str) -> String {
-    /// remove whitespace-only lines from the start/end of lines
-    fn vertical_trim(lines: Vec<String>) -> Vec<String> {
-        let mut i = 0;
-        let mut j = lines.len();
-        // first line of all-stars should be omitted
-        if !lines.is_empty() && lines[0].chars().all(|c| c == '*') {
-            i += 1;
-        }
```

Comment:
This condition may seem useless as we never consider `/****** [...] */` as doc comment.
I am not sure about `/*!******** [..] */`, should we keep this behavior.

Reply:
I think it was supposed to address the `/******` case.
If it's not relevant now, then it's better to simplify the code and remove it.